### ISSUE TITLE
size_leaf_vector enforced to 0

### ISSUE THREAD

**Issue by weidong8405** [dmlc/xgboost#4972]
I want to store multi value in tree's leaf and i noticed that xgboost have size_leaf_vector, but the code is currently not support, have many check that size_leaf_vector must be 0
my question is that what the plan about size_leaf_vector in xgboost? 
the next question is that if i want to fork a branch to modify by myself, want to deal with multi class objective use leaf vector, any details about size_leaf_vector should i remember?

**Comment by trivialfis** [comment]
@weidong8405 I'm thinking about this on my spare time (when available).  But you are welcomed to work on it.  A few things though: 
- firstly there is a misunderstanding that if we use one tree per class then there is no correlation between classes.  That's not entirely correct since we use softmax to normalize output probability across all classes.
- The change is big.  Once you have leaf vector, you will also need to modify histogram building, split evaluation etc.
- Performance hit is uncertain.
- Having vector leaf doesn't mean good result.  I read some related literature and there's no theoretical proof that says this way it can model correlation "better".

**Comment by trivialfis** [comment]
@weidong8405 Also if you are indeed working on it.  Please let me know your progress.  You can find my personal email address on github profile.

**Comment by weidong8405** [comment]
@trivialfis  thank you for reply so rapidly, currently i am working on multi classification, i understand your first answer,but use binary classification to deal with have the drawback is that it will use more trees(num_trees * num_class), that means more overhead when use the tree model in inference time.
the other one is that  i find that binary classification's performance is not as well as opencv do the multi classification

**Comment by trivialfis** [comment]
@weidong8405 You can remove that parameter if you like.  It's never used in actual code anyway.

**Comment by weidong8405** [comment]
@trivialfis  i want to use the parameter indeed, so in conclusion you think it will have many changes if want to use leaf vector?

**Comment by trivialfis** [comment]
Yup.  But certainly doable.  :-)

**Comment by trivialfis** [comment]
Reopening as we can use more inputs.

**Comment by weidong8405** [comment]
in fact i am not so care about the training speed, inference speed and online performance is more important to me, so if i do not care about cache or vector computation, the works can be much less?

**Comment by trivialfis** [comment]
Definitely lesser.

________________________________

**Comment by RAMitchell** [comment]
Its probably going to need major changes to support this parameter. You could do a proof of concept pr and see how hard it is.

**Comment by trivialfis** [comment]
Closing in prefer to this one: https://github.com/dmlc/xgboost/issues/4840
